creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+"""
+A fake KF study generator that generates realistic studies
+Used to setup unit tests for ingest runs and related functionality
+
+1. (Optional) Create study files
+- Create a biological TSV file with minimal family, participant, specimen data
+- Create source and harmonized genomic files
+- Upload genomic files to study bucket on S3
+- Create S3 object manifest for source genomic files
+- Create the BIX Genomic Workflow Output Manifest
+
+2. Ingest study files into the KF Data Service
+"""
+import os
+import logging
+import random as ra
+import string
+from pprint import pprint
+
+import boto3
+from botocore.exceptions import ClientError
+import base32_crockford as b32
+import requests
+import pandas as pd
+
+from kf_lib_data_ingest.common.io import read_df
+from kf_lib_data_ingest.common import pandas_utils
+from kf_lib_data_ingest.app import settings as ingest_settings
+from kf_lib_data_ingest.etl.ingest_pipeline import DataIngestPipeline
+from d3b_utils.aws_bucket_contents import fetch_aws_bucket_obj_info
+
+# For data generation
+ROOT_DIR = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
+DATA_DIR = os.path.join(ROOT_DIR, "data_generator", "ingest_package", "data")
+INGEST_PKG_DIR = os.path.join(ROOT_DIR, "data_generator", "ingest_package")
+
+# Connections
+DATASERVICE_URL = "http://localhost:5000"
+AWS_PROFILE = os.environ.get('AWS_PROFILE', "saml")
+TEMP_STUDY_BUCKET = "kf-api-study-creator-ingest-test"
+boto3.setup_default_session(profile_name=AWS_PROFILE)
+
+
+class StudyGenerator(object):
+
+    def __init__(
+        self,
+        data_dir=DATA_DIR,
+        ingest_package_dir=INGEST_PKG_DIR,
+        study_id="SD_ME0WME0W",
+        sequencing_centers=None,
+        total_specimens=10,
+    ):
+        self.data_dir = data_dir
+        self.study_id = study_id
+        if not sequencing_centers:
+            sequencing_centers = [
+                {
+                    "kf_id": "SC_DGDDQVYR",
+                    "name": "Broad Institute",
+                }
+            ]
+        self.sequencing_centers = sequencing_centers
+        self.total_specimens = total_specimens
+        self.source_gf_extensions = [".cram", ".crai", ".cram.md5"]
+        self.logger = logging.getLogger(type(self).__name__)
+
+        # Initialize ingest - use default settings (KF Data Service)
+        self.ingest_pipeline = DataIngestPipeline(
+            ingest_package_dir,
+            ingest_settings.load().TARGET_API_CONFIG,
+        )
+        # KF ID Prefixes
+        self.id_prefixes = {
+            "family": "FM",
+            "participant": "PT",
+            "biospecimen": "BS",
+            "genomic_file": "GF",
+        }
+
+    @property
+    def study_bucket(self):
+        """ Return study bucket from study KF ID """
+
+        return TEMP_STUDY_BUCKET
+        # return f"kf-study-us-east-1-prd-{study_id.lower().replace('_', '-')}"
```

Comment:
We need to make a study bucket in all the dev, qa, prd deployed environments for ingest testing, ingest the data files into that bucket, and then we can replace this with the commented line below.